test/integration/master/graceful_shutdown_test.go

```diff
@@ -92,6 +92,8 @@ func TestGracefulShutdown(t *testing.T) {
 	}
 	body.Close()
 	respErr := <-respErrCh
+	t.Logf("respErr %v", respErr)
+	t.Logf("respErr %v", respErr.err)
 	if err != nil {
```

Comment:
This should be `if respErr.err != nil`

Reply:
Looking at the failures, it seems to be a nil pointer somewhere around here:
```
--- FAIL: TestGracefulShutdown (12.71s)
    testserver.go:143: runtime-config=map[api/all:true]
    testserver.go:144: Starting kube-apiserver on port 43375...
    testserver.go:160: Waiting for /healthz to be ok...
    graceful_shutdown_test.go:51: server should be blocking request for data in body
    graceful_shutdown_test.go:66: server should answer
    graceful_shutdown_test.go:73: shutting down server
    graceful_shutdown_test.go:76: server should fail new requests
    graceful_shutdown_test.go:88: server should answer pending request
    graceful_shutdown_test.go:95: respErr {<nil> 0xc0492fe4b0}
    graceful_shutdown_test.go:96: respErr read tcp 127.0.0.1:43316->127.0.0.1:43375: read: connection reset by peer
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x386d408]

goroutine 50811 [running]:
testing.tRunner.func1(0xc083d65600)
	/usr/local/go/src/testing/testing.go:830 +0x392
panic(0x3ee3360, 0xaef2450)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
k8s.io/kubernetes/test/integration/master.TestGracefulShutdown(0xc083d65600)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/integration/master/graceful_shutdown_test.go:100 +0xb08
testing.tRunner(0xc083d65600, 0x48901b0)
	/usr/local/go/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x35a
FAIL	k8s.io/kubernetes/test/integration/master	194.552s
```

Reply:
I'll try this approach, thanks Mike